frontend/test/metabase/scenarios/models/reproductions/22715-remapped-values-override-column-identifier.cy.spec.js

```diff
@@ -0,0 +1,75 @@
+import {
+  restore,
+  visitQuestion,
+  popover,
+  filter,
+} from "__support__/e2e/cypress";
+
+describe.skip("issue 22715 ", () => {
+  beforeEach(() => {
+    cy.intercept("POST", "/api/dataset").as("dataset");
+    cy.intercept("PUT", "/api/card/*").as("updateModel");
+
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("fitlering based on the remapped column name should result in a correct query (metabase#22715)", () => {
+    cy.createNativeQuestion({
+      native: {
+        query: `select 1 as "ID", current_timestamp::datetime as "ALIAS_CREATED_AT"`,
+      },
+    }).then(({ body: { id } }) => {
+      // Visit the question to first load metadata
+      visitQuestion(id);
+
+      // Turn the question into a model
+      cy.request("PUT", `/api/card/${id}`, { dataset: true });
+
+      // Let's go straight to the model metadata editor
+      cy.visit(`/model/${id}/metadata`);
+
+      // The first column `ID` is automatically selected
+      mapColumnTo({ table: "Orders", column: "ID" });
+
+      cy.findByText("ALIAS_CREATED_AT").click();
+      mapColumnTo({ table: "Orders", column: "Created At" });
+
+      // Make sure the column name updated before saving
+      cy.findByDisplayValue("Created At");
+
+      cy.button("Save changes").click();
+      cy.wait("@updateModel");
+
+      cy.visit(`/model/${id}`);
+      cy.wait("@dataset");
+    });
+
+    filter();
+
+    cy.findByTestId("sidebar-right").within(() => {
+      cy.findByText("Created At").click();
+      cy.findByText("Today").click();
+    });
```

Comment:
Could we also test step5 (clicking column > filter this column) - just to make sure that works. I cannot figure out if this metadata issue is the root cause of that or if it's actually a new different issue.